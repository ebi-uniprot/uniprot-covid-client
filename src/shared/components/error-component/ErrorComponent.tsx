import React from 'react';
import { Message } from 'franklin-sites';

import './styles/error-component.scss';

const ErrorComponent = () => (
  <div className="error-component">
    <Message level="failure">
      <h5>An unexpected issue occured</h5>
      <p>
        You can try to reload the page, use the rest of this page, or go back to
        the previous page.
      </p>
      <p>
        If the error persists, please{' '}
        <a
          target="_blank"
          href="https://goo.gl/forms/VrAGbqg2XFg6Mpbh1"
          rel="noopener noreferrer"
        >
          report this bug here
        </a>
        .
      </p>
    </Message>
  </div>
);

export default ErrorComponent;
