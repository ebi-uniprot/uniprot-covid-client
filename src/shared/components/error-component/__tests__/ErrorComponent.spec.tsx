import React from 'react';
import { render } from '@testing-library/react';

import ErrorComponent from '../ErrorComponent';

describe('ErrorComponent', () => {
  test('should render', () => {
    const { asFragment } = render(<ErrorComponent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
