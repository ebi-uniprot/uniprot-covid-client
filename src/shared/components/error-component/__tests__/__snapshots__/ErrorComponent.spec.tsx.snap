// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent should render 1`] = `
<DocumentFragment>
  <div
    class="error-component"
  >
    <div
      class="message message--failure"
      role="status"
    >
      <div
        class="message__side-border"
      />
      <svg
        fill="currentColor"
        height="1.125em"
        stroke="currentColor"
        viewBox="0 0 19 18"
        width="1.125em"
      >
        <path
          d="M14.93 11.14h0L12.8 9l2.13-2.14a1.9 1.9 0 000-2.67h0l-.63-.64a1.9 1.9 0 00-2.68 0h0L9.5 5.7 7.35 3.55a1.9 1.9 0 00-2.68 0h0l-.63.64a1.9 1.9 0 000 2.67h0L6.18 9l-2.14 2.14a1.9 1.9 0 000 2.67h0l.63.64a1.9 1.9 0 002.68 0L9.5 12.3l2.13 2.14a1.9 1.9 0 002.68 0l.63-.64a1.9 1.9 0 000-2.67zM17.5 9a8 8 0 11-16 0 8 8 0 0116 0z"
          stroke-width="2"
        />
      </svg>
      <section
        class="message__content"
      >
        <small>
          <h5>
            An unexpected issue occured
          </h5>
          <p>
            You can try to reload the page, use the rest of this page, or go back to the previous page.
          </p>
          <p>
            If the error persists, please 
            <a
              href="https://goo.gl/forms/VrAGbqg2XFg6Mpbh1"
              rel="noopener noreferrer"
              target="_blank"
            >
              report this bug here
            </a>
            .
          </p>
        </small>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
